plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java

```diff
@@ -192,41 +197,42 @@ private void writeTransactionEntry(PageBuilder pageBuilder, RowType entryType, T
         appendNullOtherBlocks(pageBuilder, TXN_BLOCK_CHANNEL);
     }
 
-    private void writeAddFileEntry(PageBuilder pageBuilder, RowType entryType, AddFileEntry addFileEntry)
+    private void writeAddFileEntry(PageBuilder pageBuilder, RowType entryType, AddFileEntry addFileEntry, boolean writeStatsAsJson, boolean writeStatsAsStruct)
     {
         pageBuilder.declarePosition();
         BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(ADD_BLOCK_CHANNEL);
         BlockBuilder entryBlockBuilder = blockBuilder.beginBlockEntry();
-        writeString(entryBlockBuilder, entryType, 0, "path", addFileEntry.getPath());
-        writeStringMap(entryBlockBuilder, entryType, 1, "partitionValues", addFileEntry.getPartitionValues());
-        writeLong(entryBlockBuilder, entryType, 2, "size", addFileEntry.getSize());
-        writeLong(entryBlockBuilder, entryType, 3, "modificationTime", addFileEntry.getModificationTime());
-        writeBoolean(entryBlockBuilder, entryType, 4, "dataChange", addFileEntry.isDataChange());
-        // TODO: determine stats format in checkpoint based on table configuration; (https://github.com/trinodb/trino/issues/12031)
-        // currently if addFileEntry contains JSON stats we will write JSON
-        // stats to checkpoint and if addEntryFile contains parsed stats, we
-        // will write parsed stats to the checkpoint.
-        writeJsonStats(entryBlockBuilder, entryType, addFileEntry);
-        writeParsedStats(entryBlockBuilder, entryType, addFileEntry);
-        writeStringMap(entryBlockBuilder, entryType, 7, "tags", addFileEntry.getTags());
+        int fieldId = 0;
+        writeString(entryBlockBuilder, entryType, fieldId++, "path", addFileEntry.getPath());
+        writeStringMap(entryBlockBuilder, entryType, fieldId++, "partitionValues", addFileEntry.getPartitionValues());
+        writeLong(entryBlockBuilder, entryType, fieldId++, "size", addFileEntry.getSize());
+        writeLong(entryBlockBuilder, entryType, fieldId++, "modificationTime", addFileEntry.getModificationTime());
+        writeBoolean(entryBlockBuilder, entryType, fieldId++, "dataChange", addFileEntry.isDataChange());
+        if (writeStatsAsJson) {
+            writeJsonStats(entryBlockBuilder, entryType, addFileEntry, fieldId++);
+        }
+        if (writeStatsAsStruct) {
+            writeParsedStats(entryBlockBuilder, entryType, addFileEntry, fieldId++);
+        }
+        writeStringMap(entryBlockBuilder, entryType, fieldId++, "tags", addFileEntry.getTags());
```

Comment:
nit: the last increment for `fieldId` is not needed.

Reply:
Yes, I know. The increment is for future changes (=new entry after `tags`).